question/bank/previewquestion/preview.php

```diff
@@ -145,7 +146,7 @@
         $newoptions->variant = $options->variant;
     }
     if (isset($newoptions->saverestart)) {
-        previewquestion_helper::restart_preview($previewid, $question->id, $newoptions, $context, $returnurl);
+        previewquestion_helper::restart_preview($previewid, $newoptions->version, $newoptions, $context, $returnurl);
```

Comment:
you should not replace the questionid with version, you can add version at the end as a new parameter, id of the question is a unique identifier, it should not be resolved by version and will throw error for some qtypes